gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2

```diff
@@ -212,6 +213,17 @@ def _get_http_options():
     # Jsonify the query params
     query_params = _Base{{ service_name }}RestTransport._Base{{method_name}}._get_query_params_json(transcoded_request)
 
+    if CLIENT_LOGGING_SUPPORTED and _LOGGER.isEnabledFor(logging.DEBUG):
+        _LOGGER.debug(
+            f"Sending request via {{ service.meta.address.proto_package_versioned }}.{{ service.client_name }}.{{ method_name }}",
+            extra = {
+                "serviceName": "{{ service.meta.address.proto }}",
+                "rpcName": "{{ method_name }}",
+                "retryAttempt": 1,
+                "request": type(request).to_json(request),
```

Comment:
Does this request / metadata include request method and request url? We can extract that information out from the http_options defined above.

Reply:
Do we want the fields to just be part of `metadata` or `request`? In that case, adding these fields in google-api-core may not be required.

Adding a TODO comment here linking to an issue would be helpful to understand what's missing.

Reply:
The [structured logging fields](https://cloud.google.com/logging/docs/structured-logging#structured_logging_special_fields) include `httpRequest` which [includes](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest) `requestMethod` and `requestUrl`.

So we should populate those subfields of `httpRequest`, which is [already registered](https://github.com/googleapis/python-api-core/blob/d18d9b5131162b44eebcc0859a7aca1198a2ac06/google/api_core/client_logging.py#L13-L24) in api-core.